<launch>
    <arg name="model" default="$(find vector_ros_driver)/urdf/vector.urdf.xacro" />
    <param name="robot_description" command="$(find xacro)/xacro --inorder $(arg model)" />

    <node name="joint_state_publisher" pkg="joint_state_publisher" type="joint_state_publisher">
        <param name="publish_frequency" value="10.0"/>
        <param name="use_gui" value="false"/>
        <rosparam param="source_list">["/vector/joint_states"]</rosparam>
    </node>

    <node name="robot_state_publisher" pkg="robot_state_publisher" type="state_publisher">
        <param name="rate" value="10.0"/>
    </node>

    <arg name="ticks_per_meter" value="1000" />
    <arg name="wheel_separation" value="0.055" />

    <node name="controller" pkg="diff_drive" type="diff_drive_controller" output="screen">
        <rosparam subst_value="true">
        ticks_per_meter: $(arg ticks_per_meter)
        wheel_separation: $(arg wheel_separation)
        max_motor_speed: 3000
        timeout: 1.0
        </rosparam>
        <remap from="cmd_vel" to="/vector/cmd_vel" />
    </node>

    <node name="odom_publisher" pkg="diff_drive" type="diff_drive_odometry" output="screen">
        <rosparam subst_value="true">
          ticks_per_meter: $(arg ticks_per_meter)
          wheel_separation: $(arg wheel_separation)
        </rosparam>
    </node>

    <node name="diff_drive_go_to_goal" pkg="diff_drive" type="diff_drive_go_to_goal" output="screen">
        <param name="~rate" value="20" />
        <param name="~kP" value="0.5" />
        <param name="~kA" value="1.0" />
        <param name="~kB" value="-0.8" />
        <param name="~max_linear_speed" value="0.2" />
        <param name="~min_linear_speed" value="0.05" />
        <param name="~max_angular_speed" value="0.7" />
        <param name="~min_angular_speed" value="0.1" />
        <param name="~linear_tolerance" value="0.01" />
        <param name="~angular_tolerance" value="0.04" />
        <param name="~forwardMovementOnly" value="false" />
        <remap from="cmd_vel" to="/vector/cmd_vel" />
    </node>
</launch>